src/ImageLoaders/IHex32/IntelHex32Writer.cs

```diff
@@ -0,0 +1,215 @@
+        /// </summary>
+        /// <param name="address">The 16-bit address to write the data.</param>
+        /// <param name="data">The block data to write.</param>
+        /// <remarks>
+        /// <paramref name="data"/> can be, at most, 255 bytes (even if checksum used is not strong enough).
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">If <paramref name="data"/> is null.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">If <paramref name="data"/> size is greater than 255..</exception>
+        public void WriteData(ushort address, IList<byte> data)
+        {
+            if (data == null)
+                throw new ArgumentNullException(nameof(data));
+
+            if (data.Count > maxRecordDataSize)
+                throw new ArgumentOutOfRangeException(nameof(data), $"Must be less than {maxRecordDataSize} bytes long.");
+
+            WriteHexRecord(IntelHex32RecordType.Data, address, data);
+        }
+
+        /// <summary>
+        /// Close the file flushing pending changes to the underlying stream and writing EOF record
+        /// </summary>
+        public void Close()
+        {
+            if (!disposedValue)
+            {
+                streamWriter?.WriteLine(":00000001FF");
+                streamWriter?.Flush();
+            }
+        }
+
+        #endregion
+
+        #region Helpers
+
+        private static List<byte> FormatAddress(IntelHex32AddressType addressType, int address)
+        {
+            var result = new List<byte>();
+            var shift = (byte)(addressType == IntelHex32AddressType.ExtendedSegmentAddress ? 4 : 0);
+            shift = (byte)(addressType == IntelHex32AddressType.ExtendedLinearAddress ? 16 : shift);
+
+            var addressBytes = BitConverter.GetBytes(address >> shift);
+
+            if (addressType == IntelHex32AddressType.StartLinearAddress)
+            {
+                result.Add(addressBytes[3]);
+                result.Add(addressBytes[2]);
+            }
+
+            result.Add(addressBytes[1]);
+            result.Add(addressBytes[0]);
+
+            return result;
+        }
+
+        private static byte CalculateChecksum(IList<byte> checkSumData)
+        {
+            var maskedSumBytes = checkSumData.Sum(x => x) & 0xff;
+            var calculatedChecksum = (byte)(256 - maskedSumBytes);
+
+            return calculatedChecksum;
+        }
+
+        private static string ToHexString(IList<byte> data)
+        {
+            var result = new char[data.Count * 2];
+            for (var i = 0; i < data.Count; i++)
+            {
+                byte b = data[i];
+                result[2 * i] = HexDigits[b >> 4];
+                result[2 * i + 1] = HexDigits[b & 0xF];
+            }
+
+            return new string(result);
+        }
+        private void WriteHexRecord(IntelHex32RecordType recordType, ushort address, IList<byte> data)
```

Comment:
Nitpick: preferred formatting is a single blank line between methods.

Reply:
My mistake.